packages/medusa/src/repositories/money-amount.ts

```diff
@@ -119,40 +169,74 @@ export const MoneyAmountRepository = dataSource
           }
         }
 
-        queryBuilder.orWhere(
+        maDeleteQueryBuilder.orWhere(
           new Brackets((localQueryBuild) => {
             localQueryBuild.where(where).andWhere(orWhere)
           })
         )
       }
+      const deleteAmounts = await maDeleteQueryBuilder.getMany()
+
+      if (!deleteAmounts.length) {
+        return
+      }
 
-      await queryBuilder.execute()
+      await Promise.all([
+        this.delete(deleteAmounts.map((mav) => mav.id)),
+        mavDeleteQueryBuilder
+          .where(
+            deleteAmounts.map((mav) => ({
+              money_amount_id: mav.id,
+            }))
+          )
+          .execute(),
+      ])
     },
 
     async upsertVariantCurrencyPrice(
       variantId: string,
       price: Price
     ): Promise<MoneyAmount> {
-      let moneyAmount = await this.findOne({
-        where: {
-          currency_code: price.currency_code,
-          variant_id: variantId,
-          region_id: IsNull(),
-          price_list_id: IsNull(),
-        },
-      })
+      let moneyAmount = await this.createQueryBuilder()
+        .leftJoinAndSelect(
+          "product_variant_money_amount",
+          "mav",
+          "mav.money_amount_id = ma.id"
+        )
+        .where("mav.variant_id = :variantId", { variantId })
+        .andWhere("ma.currency_code = :currencyCode", {
+          currencyCode: price.currency_code,
+        })
+        .andWhere("ma.region_id IS NULL")
+        .andWhere("ma.price_list_id IS NULL")
+        .getOne()
+
+      const created = !moneyAmount
 
       if (!moneyAmount) {
         moneyAmount = this.create({
           ...price,
           currency_code: price.currency_code?.toLowerCase(),
-          variant_id: variantId,
+          variant: { id: variantId },
```

Comment:
comment: Does that work? maybe I ll wait the result of the last test we discussed earlier